server/src/test/java/org/elasticsearch/action/admin/indices/template/reservedstate/ReservedComposableIndexTemplateActionTests.java

```diff
@@ -877,7 +876,7 @@ public void testTemplatesWithReservedPrefix() throws Exception {
             )
             .build();
 
-        ClusterService mockedClusterService = spy(clusterService);
+        ClusterService mockedClusterService = mock(ClusterService.class);
```

Comment:
Fixes:   java.lang.IllegalArgumentException: Please don't pass mock here. Spy is not allowed on mock.

Reply:
Since `clusterService` is already a mock, we could just use the member as is, no need to create another mock?

Reply:
Code later in this method mutates the mock. I didn't want to inadvertently affect other usages.  But it did work fine. I can revert.

Reply:
The mock is created for every test, so it is already unique to this test.

Reply:
Thanks Ryan. Done.